src/mxbase.jl

```diff
@@ -42,24 +42,24 @@ function get_paths()
         end
         default_startcmd = "exec $(Base.shell_escape(default_startcmd)) -nosplash"
     elseif is_windows()
-        default_startcmd = joinpath(matlab_homepath, "bin", (Sys.WORD_SIZE == 32 ? "win32" : "win64"), "MATLAB.exe")
+        default_startcmd = joinpath(matlab_homepath, "bin", (Sys.ARCH === :i686  ? "win32" : "win64"), "MATLAB.exe")
         if !isfile(default_startcmd)
             error("The MATLAB path is invalid. Set the MATLAB_HOME evironmental variable to the MATLAB root.")
         end
         default_startcmd *= " -nosplash"
     end
 
     # Get path to MATLAB libraries
-    if is_linux()
-        matlab_library_dir = Sys.WORD_SIZE == 32 ? "glnx86" : "glnxa64"
+    matlab_library_dir = if is_linux()
+        Sys.ARCH === :i686 ? "glnx86" : "glnxa64"
     elseif is_apple()
-        matlab_library_dir = Sys.WORD_SIZE == 32 ? "maci" : "maci64"
+        Sys.ARCH === :i686 ? "maci" : "maci64"
     elseif is_windows()
-        matlab_library_dir = Sys.WORD_SIZE == 32 ? "win32" : "win64"
+        Sys.ARCH === :i686 ? "win32" : "win64"
```

Comment:
If you're saying the first binary is supposedly 32 bit, then what makes you think that's the case, and do you have any evidence leading you to believe that? For example, I see
```
Tony@LAPTOP-O230JCFF ~/julia
$ file usr/bin/julia.exe # in my local 64 bit source build
usr/bin/julia.exe: PE32+ executable (console) x86-64, for MS Windows

Tony@LAPTOP-O230JCFF ~/julia
$ file ~/julia32/usr/bin/julia.exe # local 32 bit source build
/home/Tony/julia32/usr/bin/julia.exe: PE32 executable (console) Intel 80386, for MS Windows
```
or you could look at the exe in dependency walker and I believe it should use a different icon for 32 vs 64 bit